snuba/datasets/processors/spans_processor.py

```diff
@@ -77,131 +77,88 @@ class SpansMessageProcessor(DatasetMessageProcessor):
-            return None
-
-        if not data.get("contexts", {}).get("trace"):
+        message: SpanEvent,
+    ) -> Optional[Tuple[SpanEvent, RetentionDays]]:
+        if not message.get("trace_id"):
             return None
         try:
             # We are purposely using a naive datetime here to work with the
             # rest of the codebase. We can be confident that clients are only
             # sending UTC dates.
             retention_days = enforce_retention(
-                event.get("retention_days"),
-                datetime.utcfromtimestamp(data["timestamp"]),
+                message["retention_days"],
+                datetime.utcfromtimestamp(message["start_timestamp_ms"] / 1000),
             )
         except EventTooOld:
             return None
 
-        return event, retention_days
+        return message, retention_days
 
-    def _process_base_event_values(
+    def _process_span_event(
         self,
         processed: MutableMapping[str, Any],
-        event_dict: EventDict,
-        common_span_fields: CommonSpanDict,
+        span_event: SpanEvent,
     ) -> None:
-        processed["transaction_id"] = common_span_fields["transaction_id"] = str(
-            uuid.UUID(event_dict["event_id"])
-        )
-
-        transaction_ctx = event_dict["data"]["contexts"]["trace"]
-        trace_id = transaction_ctx["trace_id"]
-        processed["trace_id"] = str(uuid.UUID(trace_id))
-        processed["span_id"] = int(transaction_ctx["span_id"], 16)
-        processed["segment_id"] = common_span_fields["segment_id"] = processed[
-            "span_id"
-        ]
-        processed["is_segment"] = 1
-        parent_span_id: str = transaction_ctx.get("parent_span_id", "0") or "0"
-        processed["parent_span_id"] = int(parent_span_id, 16) if parent_span_id else 0
-
-        processed["description"] = _unicodify(event_dict.get("description", ""))
-        processed["op"] = common_span_fields["transaction_op"] = _unicodify(
-            transaction_ctx.get("op", "")
-        )
-        processed["transaction_op"] = processed["op"]
-
-        span_hash = transaction_ctx.get("hash", None)
-        processed["group_raw"] = 0 if not span_hash else int(span_hash, 16)
-        processed["segment_name"] = _unicodify(
-            event_dict["data"].get("transaction") or ""
-        )
-
+        # process ids
+        processed["trace_id"] = str(uuid.UUID(span_event["trace_id"]))
+        processed["span_id"] = int(span_event["span_id"], 16)
+        processed["segment_id"] = processed["span_id"]
+        processed["is_segment"] = span_event["is_segment"]
+        parent_span_id: Optional[str] = span_event.get("parent_span_id", None)
+        if parent_span_id:
+            processed["parent_span_id"] = int(parent_span_id, 16)
+        transaction_id: Optional[str] = span_event.get("event_id", None)
+        if transaction_id:
+            processed["transaction_id"] = str(uuid.UUID(transaction_id))
```

Comment:
We're not going to have the `transaction_id` anymore, are we?

Reply:
Looks like it. If the `event_id` is supplied we use it otherwise its null.

Reply:
I believe this was needed to link spans to transactions since at one point there was a requirement to link spans to their transactions. We can followup in a separate PR if that requirement is still true or not.